### ISSUE TITLE
Mono Audio Checkbox

### ISSUE THREAD

**Issue by gigapixelguy** [VampireChicken12/youtube-enhancer#257]
Sometimes with old movie uploads or certain files have messed up stereo tracks on youtube (for example only have sound on one side)
Example of messed up audio: https://www.youtube.com/watch?v=Mh2GQmfN-AI

Either in the volume category or the misc category, add a checkbox to make mono audio (I am not sure if having it be global or video specific one time click would be better for user experience).

Basically to implement this, take all the sounds from channels, add then together, then divide by number of channels, that is the new output of all the channels.

**Comment by VampireChicken12** [comment]
> @gigapixelguy You can’t even imagine how labor-intensive this is. As far as I know, there is no way to activate this functionality in the youtube stock player.
> I was working on a module for downloading videos from Youtube. And getting the file is not at all easy.
> 1. You can get the audio file track separately,
> 2. Next pass the link to the file to this code https://stackoverflow.com/questions/69726033/javascript-new-audiocontext-mono-to-stereo-not-wordking
> It is impossible to find out which channel is being used. Often right but not always -  https://www.youtube.com/watch?v=e8ODm-F9-IM
> Therefore, it is obvious that you need to use the built-in functions of your OS or audio card
> a) https://www.reddit.com/r/youtube/comments/l5f0qo/force_audio_to_play_as_mono/
> b) https://superuser.com/questions/351249/how-do-i-force-single-only-channel-audio-in-youtube-flash-videos-to-play-in-bo
> 
> From my experience, some users have bad contact on the audio output connector. For example, if there is no Ground, then only the difference between the channels will sound. If possible, it is better to output the sound digitally -HDMI,S/PDIF
> 
> ![download](https://github.com/VampireChicken12/youtube-enhancer/assets/13064767/ce389663-cfeb-4cc5-ace5-b9c89a395016)
> 

Thanks for the information.

**Comment by danielwerg** [comment]
## workarounds

### firefox add-on

https://addons.mozilla.org/en-US/firefox/addon/soundfixer/

### linux easy effects

Presets -> New preset name (mono) -> Create new Preset -> Select new preset and close submenu -> Output -> Players -> Add Effect -> Stereo Tools -> Stereo Matrix -> Mode -> `LR > L+R (Mono sum L+R)`

`audio-mono-toggle`:

```bash
#!/bin/bash

if [ "$(easyeffects --last-loaded-preset output)" == "mono" ]; then
  easyeffects --load-preset default
  notify-send "Audio" "Stereo"
  exit 0
else
  easyeffects --load-preset mono
  notify-send "Audio" "Mono"
  exit 1
fi
```

can't remember whether or not you need to create `default` preset or it's already there.

**Comment by VampireChicken12** [comment]
<img width="162" height="115" alt="Image" src="https://github.com/user-attachments/assets/61d6b612-b902-4512-986b-cfdcd92260d5" />
<br />
<img width="107" height="122" alt="Image" src="https://github.com/user-attachments/assets/a0fe3e80-18bc-4416-a15c-eab6332161a2" />
<br />
<img width="507" height="77" alt="Image" src="https://github.com/user-attachments/assets/4d560504-9ea1-410f-aae3-30c1d5d3c937" />

**Comment by VampireChicken12** [comment]
Funny story, the only reason I worked on this feature was because a friend recently sent me a YouTube video with audio in one channel and It was annoying so I decided to fix it 🤣

**Comment by VampireChicken12** [comment]
@all-contributors please add @HALL4Y for translations
